python/ray/_private/node.py

```diff
@@ -445,7 +449,9 @@ def _init_temp(self):
             )
             self._temp_dir = self._ray_params.temp_dir
         else:
-            if self._ray_params.temp_dir is None:
+            if self._ray_params.temp_dir is not None:
+                self._temp_dir = self._ray_params.temp_dir
+            elif self._head_os == sys.platform:
                 assert not self._default_worker
                 temp_dir = ray._private.utils.internal_kv_get_with_retry(
                     self.get_gcs_client(),
```

Comment:
Now, when determining `temp_dir` location, head's `temp_dir` value is only used when worker and head are of same os. If not, the worker will fallback to use it's system default using `ray._private.utils.get_ray_temp_dir()` like head node.